// bod/dist/commands/InfoCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const utils_1 = require("../utils");
const BaseCommand_1 = tslib_1.__importDefault(require("./BaseCommand"));
class InfoCommand extends BaseCommand_1.default {
    constructor() {
        super({
            name: 'info',
            description: 'Print debugging information about your environment',
            usage: 'info',
        });
    }
    run(_appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            utils_1.printer.info(`Environment information:`);
            const envInfo = yield utils_1.envinfo.run({
                System: ['OS', 'CPU'],
                Binaries: ['Node', 'Yarn', 'npm'],
                Browsers: ['Chrome', 'Edge', 'Firefox', 'Safari'],
                npmPackages: '/**/{typescript,*react*,@sabertazimi/*/}',
                npmGlobalPackages: ['@sabertazimi/bod'],
            }, {
                showNotFound: true,
                duplicates: true,
                fullTree: true,
            });
            utils_1.printer.info(envInfo);
        });
    }
}
exports.default = InfoCommand;
